library/spdm_responder_lib/libspdm_rsp_measurement_extension_log.c

```diff
@@ -0,0 +1,145 @@
+        return libspdm_responder_handle_response_state(
+            spdm_context,
+            spdm_request->header.request_response_code,
+            response_size, response);
+    }
+    if (spdm_context->connection_info.connection_state < LIBSPDM_CONNECTION_STATE_NEGOTIATED) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_UNEXPECTED_REQUEST,
+                                               0, response_size, response);
+    }
+
+    session_info = NULL;
+    if (spdm_context->last_spdm_request_session_id_valid) {
+        session_info = libspdm_get_session_info_via_session_id(
+            spdm_context,
+            spdm_context->last_spdm_request_session_id);
+        if (session_info == NULL) {
+            return libspdm_generate_error_response(
+                spdm_context,
+                SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+                response_size, response);
+        }
+        session_state = libspdm_secured_message_get_session_state(
+            session_info->secured_message_context);
+        if (session_state != LIBSPDM_SESSION_STATE_ESTABLISHED) {
+            return libspdm_generate_error_response(
+                spdm_context,
+                SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+                response_size, response);
+        }
+    }
+    if (!libspdm_is_capabilities_flag_supported(
+            spdm_context, false, 0,
+            SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEL_CAP)) {
+        return libspdm_generate_error_response(
+            spdm_context, SPDM_ERROR_CODE_UNSUPPORTED_REQUEST,
+            0, response_size, response);
+    }
+
+    if (request_size < sizeof(spdm_get_measurement_extension_log_request_t)) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_INVALID_REQUEST, 0,
+                                               response_size, response);
+    }
+
+    offset = spdm_request->offset;
+    length = spdm_request->length;
+
+    if (!libspdm_is_capabilities_flag_supported(spdm_context, false,
+                                                SPDM_GET_CAPABILITIES_REQUEST_FLAGS_CHUNK_CAP,
+                                                SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP)) {
+        if (length > LIBSPDM_MAX_MEL_BLOCK_LEN) {
+            length = LIBSPDM_MAX_MEL_BLOCK_LEN;
+        }
+    }
+
+    spdm_mel = NULL;
+    spdm_mel_len = 0;
+    if (!libspdm_measurement_extension_log_collection(
+            spdm_context,
+            spdm_context->connection_info.algorithm.mel_spec,
+            spdm_context->connection_info.algorithm.measurement_spec,
+            spdm_context->connection_info.algorithm.measurement_hash_algo,
+            (void **)&spdm_mel, &spdm_mel_len)) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_INVALID_REQUEST, 0,
```

Comment:
`SPDM_ERROR_CODE_INVALID_REQUEST` -> `SPDM_ERROR_CODE_OPERATION_FAILED`

Reply:
I have changed the return error. Thanks.